### PR TITLE
Add additional OTEL tracing into grpc/oob/bmc/bmc.go

### PR DESCRIPTION
**Pull request by nshalman**
## Description

Create client.Thing spans in bmc.go functions that do the heavy lifting.

## Why is this needed

Fixes: #122 

## How Has This Been Tested?

- [x] Manually

## How are existing users impacted? What migration steps/scripts do we need?

If you're sending otel spans, now you'll send more.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.16.9 as builder
+FROM golang:1.17.11 as builder
```
Comment (nshalman) [inline Dockerfile:1]: Should this be a separate PR?

**Review by nshalman — COMMENTED** [review]
Ideally I'd love to get the lower level bmclib information like I do for the power and boot device code paths, but this feels like a good start to at least reveal these functions running.

**Comment by nshalman** [comment]
@tobert I'm also interested in your feedback even though GitHub won't let me add you as a reviewer.

File: Dockerfile (hunk 1 above)
Comment (jacobweinstock) [reply Dockerfile:1]: i dont think so. it updates to match the version in CI. https://github.com/tinkerbell/pbnj/blob/34853cdbf069ad1efbfd4babba9fd0464e7c08ec/.github/workflows/ci.yaml#L17

File: grpc/oob/bmc/bmc.go [hunk 2]
```diff
@@ -207,27 +234,32 @@ func (m Action) UpdateUser(ctx context.Context) error {
...
+	span.SetAttributes(attribute.String("bmc.host", host), attribute.String("bmc.username", user))
 
 	creds := &v1.UserCreds{Username: m.DeleteUserRequest.Username}
 	status := fmt.Sprintf("deleting user %q", creds.GetUsername())
 	m.SendStatusMessage(status)
 
 	actions, err := m.setupConnection(ctx, user, password, host, creds)
 	if err != nil {
-		m.SendStatusMessage("connectiion setup failed")
+		// setupConnection is responsible for sending a status message and updating the span
```
Comment (jacobweinstock) [inline grpc/oob/bmc/bmc.go:256]: don't we still want to send a `m.SendStatusMessage()` here?

File: grpc/oob/bmc/bmc.go [hunk 3]
```diff
@@ -207,27 +234,32 @@ func (m Action) UpdateUser(ctx context.Context) error {
...
 		return err
 	}
 
-	if err = oob.DeleteUser(ctx, actions); err != nil {
-		m.SendStatusMessage(fmt.Sprintf("error %s: %v", status, err))
-		m.Log.Info(fmt.Sprintf("error %s: %v", status, err))
+	err = oob.DeleteUser(ctx, actions)
+	if err != nil {
+		m.noteError(fmt.Sprintf("error %s: %v", status, err), span)
```
Comment (jacobweinstock) [inline grpc/oob/bmc/bmc.go:262]: curious, what's the driving force to move away from the one line if statement?

File: grpc/oob/bmc/bmc.go [hunk 4]
```diff
@@ -144,30 +152,44 @@ func (m Action) setupConnection(ctx context.Context, user, password, host string
...
+	span.SetAttributes(attribute.String("bmc.host", host), attribute.String("bmc.username", user))
 
 	creds := m.CreateUserRequest.GetUserCreds()
 	status := fmt.Sprintf("creating user %q", creds.GetUsername())
 	m.SendStatusMessage(status)
 
 	actions, err := m.setupConnection(ctx, user, password, host, creds)
 	if err != nil {
-		m.SendStatusMessage("connection setup failed")
+		// setupConnection is responsible for sending a status message and updating the span
```
Comment (jacobweinstock) [inline grpc/oob/bmc/bmc.go:186]: don't we still want to send a `m.SendStatusMessage()` here?

File: grpc/oob/bmc/bmc.go (hunk 2 above)
Comment (nshalman) [reply grpc/oob/bmc/bmc.go:256]: There's a SendStatusMessage in setupConnection. Seems duplicative to send two messages where one should suffice.

File: grpc/oob/bmc/bmc.go (hunk 3 above)
Comment (nshalman) [reply grpc/oob/bmc/bmc.go:262]: I think other lines had it split. I can move all three to one line.

Comment (nshalman) [reply grpc/oob/bmc/bmc.go]: Cleaned up in all three locations.

File: Dockerfile (hunk 1 above)
Comment (nshalman) [reply Dockerfile:1]: Will keep it in here. Thanks!

File: grpc/oob/bmc/bmc.go (hunk 2 above)
Comment (nshalman) [reply grpc/oob/bmc/bmc.go:254]: I could also move the comment above the function call and consolidate this into a 1 line `if` statement... Thoughts?

File: grpc/oob/bmc/bmc.go (hunk 4 above)
Comment (jacobweinstock) [reply grpc/oob/bmc/bmc.go:186]: ah ok, never mind. this is good, thanks

**Review by jacobweinstock — APPROVED** [review]
